### PR TITLE
improve(truffle): upgrade to latest truffle version

### PR DESCRIPTION
**Pull request by pemulis**
Signed-off-by: John Shutt <john.d.shutt@gmail.com>

**Motivation**

Upgrades Truffle to v5.1.63 so that it works with the latest versions of nodejs.

**Summary**

Update package.json and `yarn build`.

**Details**

Ran into some difficulty due to this when doing Kovan migrations.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [x]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

N/A

### CODE REVIEW

**Comment by pemulis** [comment]
Marking this as ready for review since it looks like the Slither failure is not specific to the Truffle update, and is being addressed in https://github.com/UMAprotocol/protocol/pull/2478. Not sure how we want to handle PRs in the meantime, since it seems none of them will make it past Slither.

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "eslint-plugin-react": "^7.19.0",
     "eslint-plugin-react-hooks": "^3.0.0",
     "ganache-cli": "^6.12.2",
+    "hardhat": "^2.0.8",
```
Comment (pemulis) [inline package.json:43]: I installed `hardhat` to try to fix the Slither issue, but it didn't work. This dependency might not be necessary after all.

Comment (mrice32) [reply package.json:43]: I could see it being useful to have it be a top-level dev dependency and/or being a devDependency of every package where it isn't needed as a runtime dep.

**Review by mrice32 — COMMENTED** [review]
@chrismaree Where do you think the hardhat binary should be installed? As a top-level dev dependency? Or a package-by-package dev dependency?

**Comment by chrismaree** [comment]
> @chrismaree Where do you think the hardhat binary should be installed? As a top-level dev dependency? Or a package-by-package dev dependency?

ye, the PR where I was trying this https://github.com/UMAprotocol/protocol/pull/2478 was just an experiment based on the error message generated in CI. Still need to explore more what's going on exactly.

**Comment by mrice32** [comment]
@pemulis do we still want to merge this or is it unnecessary now?

**Comment by pemulis** [comment]
> @pemulis do we still want to merge this or is it unnecessary now?

I think it's unnecessary. We could look at doing this as part of a broader upgrade of our packages.

**Pull request closed without merging**